File: app/src/main/java/com/davidbronn/fragmentrevision/inter_fragments/SimpleViewPagerAdapter.java
package com.davidbronn.fragmentrevision.inter_fragments;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class SimpleViewPagerAdapter extends FragmentPagerAdapter
{
    SimpleViewPagerAdapter(FragmentManager fm)
    {
        super(fm);
    }

    @Override
    public Fragment getItem(int position)
    {
        switch (position)
        {
            case 0:
                return new FirstFragment();
            case 1:
                return new SecondFragment();
            default:
                return null;
        }
    }

    @Override
    public int getCount()
    {
        return 2;
    }
}

File: app/src/main/java/com/davidbronn/fragmentrevision/withinViewPager/SimpleViewPagerAdapter.kt
package com.davidbronn.fragmentrevision.withinViewPager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class SimpleViewPagerAdapter internal constructor(fm: FragmentManager)
    : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> FirstFragment()
            1 -> SecondFragment()
            else -> FirstFragment()
        }
    }

    override fun getCount(): Int {
        return 2
    }
}
